src/brevitas/graph/gpxq.py

```diff
@@ -98,6 +98,8 @@ def __enter__(self):
 
                 # Attach hooks for GPTQ
                 if self._is_module_supported(module):
+                    # TODO: create init_class_something method, that each class implements and
+                    # where the class specific arguments will be specified
                     gpxq = self.class_implementation(
```

Comment:
Instead of calling self.class_implementation, each "mode" should implement `def init_class`.
This functions accepts module, name, parallel_layer, and all the arguments that are shared across GPTQ, GPFQ, GPFQA2Q.

gptq_mode will implement this method by inizializing and returning a GPTQ instance with the extra arguments needed for GPTQ (i.e., num_blocks).

Similarly for gpfq_mode